### PR TITLE
Proposed model

### PR DESCRIPTION
**Pull request by lymereJ**
The proposed changes add the capability to generate a proposed building model from a user model in addition to the baseline model following the ASHRAE Standard 90.1-Appendix G rules.

### CODE REVIEW

File: lib/openstudio-standards/standards/Standards.PlantLoop.rb [hunk 1]
```diff
@@ -1476,4 +1476,43 @@ def plant_loop_capacity_w_by_maxflow_and_delta_t_forwater(plant_loop)
...
+      pipe = nil
+      case obj_type
+      when 'OS_Pipe_Indoor'
+        pipe = component.to_PipeIndoor.get
+      when 'OS_Pipe_Outdoor'
+        pipe = component.to_PipeOutdoor.get
+      end
+
+      # Get pipe node
+      node = pipe.to_StraightComponent.get.outletModelObject.get.to_Node.get
```
[the remark this answers is not in the document]
Comment (lymereJ) [reply lib/openstudio-standards/standards/Standards.PlantLoop.rb]: Fixed!

File: lib/openstudio-standards/standards/Standards.Model.rb [hunk 2]
```diff
@@ -65,11 +65,14 @@ def model_create_prm_baseline_building(model, building_type, climate_zone, custo
   # @param debug [Boolean] If true, will report out more detailed debugging output
   # @return [Bool] returns true if successful, false if not
   def model_create_prm_any_baseline_building(user_model, building_type, climate_zone, hvac_building_type = 'All others', wwr_building_type = 'All others', swh_building_type = 'All others', model_deep_copy = false, custom = nil, sizing_run_dir = Dir.pwd, run_all_orients = false, unmet_load_hours_check = true, debug = false)
+    # Generate proposed model from the user-provided model
+    proposed_model = model_create_prm_proposed_building(user_model)
```
[the remark this answers is not in the document]
Comment (lymereJ) [reply lib/openstudio-standards/standards/Standards.Model.rb:81]: Addressed!

File: lib/openstudio-standards/standards/Standards.Model.rb [hunk 3]
```diff
@@ -64,18 +65,40 @@ def model_create_prm_baseline_building(model, building_type, climate_zone, custo
...
+                           "The user model is not a valid OpenStudio model. Baseline and proposed model(s) won't be created.")
+        prm_raise(false,
+                  sizing_run_dir,
+                  "The user model is not a valid OpenStudio model. Baseline and proposed model(s) won't be created.")
+      end
+
+      # Check if proposed HVAC system is autosized
+      if model_is_hvac_autosized(user_model)
+        OpenStudio.logFree(OpenStudio::Warn, 'prm.log',
+                           "The user model's HVAC system is partly autosized. Baseline and proposed model(s) won't be created.")
```
Comment (weilixu) [inline lib/openstudio-standards/standards/Standards.Model.rb:84]: Should we update this message since we will still produce baseline and proposed model?

**Review by weilixu — APPROVED** [review]
Thanks, this looks good to me.
One comment about the warning message.

**Pull request merged**